fix various small typos (#2507)
iterative/dvc.org: content/docs/start/data-and-model-versioning.md

# Get Started: Data Versioning

> How cool would it be to make Git handle arbitrary large files and directories

Reviewer: ```suggestion
How cool would it be to make Git handle arbitrarily large files and directories
```

> with the same performance that you get with small code files? Imagine doing a `git clone`

Reviewer: I agree that the current text could be improved but I'm not sure that the proposed change is much better.

Reviewer: Also the paragraph wrapping (80 chars) is broken here 🙂 (no big deal since we can merge the Restyled PR #2508).

> with the same performance that you get with small code files? Imagine doing a `git clone`

Reviewer: ```suggestion
with the same performance it has with small code files? Imagine doing a `git clone`
```
